sys/net/gnrc/network_layer/ipv6/nib/nib_ft.c

```diff
@@ -36,12 +36,18 @@ int gnrc_ipv6_nib_ft_get(const ipv6_addr_t *dst, gnrc_pktsnip_t *pkt,
 
 int gnrc_ipv6_nib_ft_add(const ipv6_addr_t *dst, unsigned dst_len,
                          const ipv6_addr_t *next_hop, unsigned iface,
-                         uint16_t ltime)
+                         uint32_t ltime)
 {
     int res = 0;
     bool is_default_route = ((dst == NULL) || (dst_len == 0) ||
                              ipv6_addr_is_unspecified(dst));
 
+    uint32_t ltime_ms = ltime * MS_PER_SEC;
+    if (ltime_ms / MS_PER_SEC != ltime) {
+        /* overflow */
+        assert(false);
```

Comment:
Since this is caused by a (valid) network packet, should we really panic here?

Reply:
Maybe a bit overreacting but I thought it could catch some misusage. 
Instead, I'd better move this:
```
    if (route_ltime < UINT32_MAX) { /* UINT32_MAX means infinite lifetime */
        /* the valid lifetime is given in seconds, but our timers work in
         * milliseconds, so we have to scale down to the smallest possible
         * value (UINT32_MAX - 1). This is however alright since we ask for
         * a new router advertisement before this timeout expires */
        route_ltime = (route_ltime > (UINT32_MAX / MS_PER_SEC)) ?
                      (UINT32_MAX - 1) : route_ltime;
    }
```

into the function then.